pkg/replica/replica.go

```diff
@@ -1265,7 +1265,12 @@ func (r *Replica) UnmapAt(length uint32, offset int64) (n int, err error) {
 		// For list `unmappableDisks`, the first entry is the volume head,
 		// the second one is the parent of the first entry...
 		unmappableDisks := []string{r.diskPath(r.activeDiskData[len(r.activeDiskData)-1].Name)}
-		for idx := len(r.activeDiskData) - 2; idx > 0; idx-- {
+		indexOfVolumeHeadParent := len(r.activeDiskData) - 2
+		indexOfLastSnapshotDiskFile := 1
+		if r.isBackingFile(int(backingFileIndex)) {
+			indexOfLastSnapshotDiskFile = int(backingFileIndex) + 1
+		}
+		for idx := indexOfVolumeHeadParent; idx >= indexOfLastSnapshotDiskFile; idx-- {
 			disk := r.activeDiskData[idx]
```

Comment:
Do you know why we keep the first of activeDiskData is empty? any context for this current design?

Reply:
No idea from the [commit history](https://github.com/longhorn/longhorn-engine/commit/cb06cb8af65ace8e4b9a861f80a6301bf91a679c#diff-6e6694d5b35c75c9d0850d82cbd57a1a0bf3f9fac601cb25f7dacfa8682fe0a5R121). Keeping the first entry empry seems not necessary from the implementation perspsective.

Reply:
yeah, very confusing. Let's keep it as is but can refactor it later.